lib/src/main/java/io/ably/lib/transport/WebSocketTransport.java

```diff
@@ -147,8 +150,29 @@ class WsClient extends WebSocketClient {
         @Override
         public void onOpen(ServerHandshake handshakedata) {
             Log.d(TAG, "onOpen()");
-            connectListener.onTransportAvailable(WebSocketTransport.this);
-            flagActivity();
+            if (isHostnameVerified(params.host)) {
+                connectListener.onTransportAvailable(WebSocketTransport.this);
+                flagActivity();
+            } else {
+                connectListener.onTransportUnavailable(WebSocketTransport.this, ConnectionManager.REASON_REFUSED);
+                close();
```

Comment:
@QuintinWillison I'm not sure if we should call this `close()` here as it will result in calling the `connectListener.onTransportUnavailable()` once again from withing `onClose()` method. WDYT?

Reply:
I think I would drop the call to `onTransportUnavailable` and just let `close` do the right thing. I note that you already have a `Log.e` statement in the `isHostnameVerified` implementation, so there should be a clue left when this happens.

Reply:
https://github.com/ably/ably-java/pull/808/commits/2999d2a48e31b9fe2fe1175281c887419c5c7f66